sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/StoreReader.java

```diff
@@ -687,6 +682,15 @@ private Mono<ReadReplicaResult> readPrimaryInternalAsync(
                         true,
                         primaryUriReference.get(),
                         replicaStatusList);
+
+                if (storeTaskException instanceof CosmosException) {
+                    CosmosException cosmosException = (CosmosException) storeTaskException;
+
+                    if (com.azure.cosmos.implementation.Exceptions.isAvoidQuorumSelectionException(cosmosException)) {
```

Comment:
```suggestion
                    if (Exceptions.isAvoidQuorumSelectionException(cosmosException)) {
```

Reply:
There is a namespace collision with `reactor.core.Exceptions` otherwise. Nevertheless, used a static import of `isAvoidQuorumSelectionException`.